src/neuroconv/tools/nwb_helpers/_models/_base_dataset_models.py

```diff
@@ -0,0 +1,131 @@
+
+    dataset_info: DatasetInfo = Field(description="The immutable information about this dataset.")
+    chunk_shape: Tuple[int, ...] = Field(  # When using Pydantic v2.0, specify PositiveInt
+        description=(
+            "The specified shape to use when chunking the dataset. "
+            "For optimized streaming speeds, a total size of around 10 MB is recommended."
+        )
+    )
+    buffer_shape: Tuple[int, ...] = Field(
+        description=(
+            "The specified shape to use when iteratively loading data into memory while writing the dataset. "
+            "For optimized writing speeds and minimal RAM usage, a total size of around 1 GB is recommended."
+        )
+    )
+    # TODO: When using Pydantic v2, wrap h5py._hl.filters.FilterRefBase and numcodecs.abc.Codec with InstanceOf
+    compression_method: Union[str, h5py._hl.filters.FilterRefBase, numcodecs.abc.Codec, None] = Field(
+        default="gzip",
+        description="The specified compression method to apply to this dataset. Set to `None` to disable compression.",
+    )
+    compression_options: Union[Dict[str, Any], None] = Field(
+        default=None, description="The optional parameters to use for the specified compression method."
+    )
+
+    def __str__(self) -> str:
+        """
+        Not overriding __repr__ as this is intended to render only when wrapped in print().
+
+        Reason being two-fold; a standard `repr` is intended to be slightly more machine readable / a more basic
+        representation of the true object state. But then also because an iterable of these objects, such as a
+        `List[DatasetConfiguration]`, would print out the nested representations, which only look good when using the
+        basic `repr` (that is, this fancy string print-out does not look good when nested in another container).
+        """
+        string = (
+            f"\n{self.dataset_info.location}"
+            f"\n{'-' * len(self.dataset_info.location)}"
+            f"\n  maxshape: {self.dataset_info.full_shape}"
+            f"\n  dtype: {self.dataset_info.dtype}"
+            "\n"
+            f"\n  chunk_shape: {self.chunk_shape}"
+            f"\n  buffer_shape: {self.buffer_shape}"
+            f"\n  compression_method: {self.compression_method}"
+        )
+        if self.compression_options is not None:
+            string += f"\n  compression_options: {self.compression_options}"
+
+        return string
+
+    @root_validator
+    def validate_all_shapes(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        chunk_shape = values["chunk_shape"]
+        buffer_shape = values["buffer_shape"]
+        full_shape = values["dataset_info"].full_shape
+
+        if len(chunk_shape) != len(buffer_shape):
+            raise ValueError(f"{len(chunk_shape)=} does not match {len(buffer_shape)=}!")
+        if len(buffer_shape) != len(full_shape):
+            raise ValueError(f"{len(buffer_shape)=} does not match {len(full_shape)=}!")
```

Comment:
```suggestion
            raise ValueError(f"{len(buffer_shape)=} does not match {len(full_shape)=}!  for dataset located in {self.dataset_info.location} ")
```

When running a transformation with multiple datasets is not going to be obvious which dataset is causing the problems. We should add the information of the dataset to the user so they know where exactly to fix this. We should do the same for the other assertions here.

Reply:
Done